docs/contributing.rst

```diff
@@ -1,7 +1,72 @@
 Contributing to sparse
 ======================
 
+General Guidelines
+------------------
 sparse is a community-driven project on GitHub. You can find our
 `repository on GitHub <https://github.com/mrocklin/sparse>`_. Feel
 free to open issues for new features or bugs, or open a pull request
 to fix a bug or add a new feature.
+
+If you haven't contributed to open-source before, we recommend you read
+`this excellent guide by GitHub on how to contribute to open source
+<https://opensource.guide/how-to-contribute/>`_. The guide is long,
+so you can gloss over things you're familiar with.
+
+Guidelines Specific to this Project
+-----------------------------------
+If you're not already familiar with it, we follow the `fork and pull model
+<https://help.github.com/articles/about-collaborative-development-models/>`_
+on GitHub.
+
+To get started, you can do something along the lines of the following, after
+creating a fork:
+
+.. code-block:: bash
+   git pull https://github.com/<username>/sparse.git
+   git remote add upstream https://github.com/mrocklin/sparse.git
+   git install -e .[docs, tests]
```

Comment:
git->pip

Reply:
However I'm also somewhat concerned about having basic github usage documentation in our docs.  I would like to reserve the right to delete all of this in the future if it becomes something we have to maintain.

Reply:
Good point, I'll remove that, and fix that.